### PR TITLE
[screeps] Publish v3.0.0

### PR DESCRIPTION
**Pull request by pyrodogg**
This PR publishes typed-screeps v3.0.0 to DefinitelyTyped.

This is a major release with breaking changes, please view the release
notes for full details https://github.com/screepers/typed-screeps/releases/tag/v3.0.0

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://screeps.com/forum/topic/2814/factories-new-resources-npc-strongholds 
  - https://github.com/screepers/typed-screeps/releases/tag/v3.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/screeps/index.d.ts [hunk 1]
```diff
@@ -4143,7 +4577,41 @@ interface SpawnOptions {
     directions?: DirectionConstant[];
 }
 
-interface SpawningConstructor extends _Constructor<Spawning>, _ConstructorById<Spawning> { }
+interface SpawningConstructor extends _Constructor<Spawning>, _ConstructorById<Spawning> {}
+interface StoreBase<POSSIBLE_RESSOURCES extends ResourceConstant, UNLIMITED_STORE extends boolean> {
```
Comment (scaytrase) [inline types/screeps/index.d.ts]: Not sure it's important, but here is a typo in POSSIBLE_RES~S~OURCES

Comment (pyrodogg) [reply types/screeps/index.d.ts]: @scaytrase Thanks. It doesn't seem critical, it's used consistently and would only be referenced by users if they're extending the interface and referencing the type variable (rarely expected). I'll take a look at updating it in a patch release.

File: types/screeps/index.d.ts [hunk 2]
```diff
@@ -1757,9 +1980,19 @@ interface _Constructor<T> {
 }
 
 interface _ConstructorById<T> extends _Constructor<T> {
-    new (id: string): T;
-    (id: string): T;
+    new (id: Id<T>): T;
+    (id: Id<T>): T;
+}
+
+// tslint:disable-next-line: no-namespace
```
Comment (amcasey) [inline types/screeps/index.d.ts]: Why disable these lint rules?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/screeps/index.d.ts (hunk 2 above)
Comment (pyrodogg) [reply types/screeps/index.d.ts]: On review they are unnecessary and have been removed in latest commit. Thanks.

File: types/screeps/index.d.ts (hunk 1 above)
Comment (pyrodogg) [reply types/screeps/index.d.ts]: Typo fixed in latest commit.

**Review by amcasey — APPROVED** [review]
Thanks!

**Pull request merged**